### PR TITLE
Embedding engine hardcodes last-token pooling, producing wrong vectors for CLS/mean models (e.g. bge-m3)

### LINKED ISSUE #89: Embedding engine hardcodes last-token pooling, producing wrong vectors for CLS/mean models (e.g. bge-m3)
## Problem

`src/engine/optimum/optimum_emb.py` always applies last-token pooling in `generate_embeddings`:

```python
embeddings = self.last_token_pool(outputs.last_hidden_state, batch_dict["attention_mask"])
```

This is correct for decoder-style embedding models like `Qwen3-Embedding-*`, but wrong for the much larger family of encoder-style models that ship via sentence-transformers — most notably:

- `BAAI/bge-m3`, `BAAI/bge-large-en-v1.5`, etc. → **CLS pooling**
- `sentence-transformers/all-MiniLM-L6-v2`, `intfloat/multilingual-e5-*` → **mean pooling**

Loading any of these through OpenArc's optimum engine today returns numerically valid but semantically wrong vectors (last hidden state of the final non-pad token instead of `[CLS]` / masked mean). There's no error and no warning — retrieval quality just silently collapses.

## Reproducer

1. Convert `BAAI/bge-m3` to OpenVINO IR via `optimum-intel` (preserves the `1_Pooling/config.json` shipped with the model).
2. Register it in `openarc_config.json` with `engine: "optimum"`, `model_type: "emb"`.
3. Hit `/v1/embeddings` and compare against the PyTorch reference — cosine similarity is ~0.3–0.6 instead of ~1.0.

## Proposal

Make pooling mode metadata-driven, with the same precedence sentence-transformers itself uses:

1. **`runtime_config.pool_mode`** (operator override; one of `"cls" | "mean" | "last"`). Unknown values raise at load time so typos don't silently fall back.
2. **`<model_path>/1_Pooling/config.json`** (auto-detect from the file the model ships with — `pooling_mode_cls_token` → cls, `pooling_mode_mean_tokens` → mean).
3. **Default**: `"last"` (preserves current Qwen3-Embedding behavior; no change for existing users).

This keeps the registry config minimal for the common case (correct pooling auto-detected from model files) while giving a clear escape hatch for models that ship without sentence-transformers metadata or with the wrong metadata.

## Scope

- `src/engine/optimum/optimum_emb.py`: add `cls_pool` / `mean_pool`, dispatch in `pool()`, resolve mode in `load_model`.
- Unit tests for each pool, `_detect_pool_mode`, the override path, and unknown-value rejection.
- Integration test loading bge-m3 and asserting cls auto-detect + 1024-dim unit-normed vector.

Happy to send a PR — branch is ready (`feat/embedding-pool-dispatch` on KIntegrated/OpenArc), verified end-to-end against PyTorch (cos > 0.999) and live via `/v1/embeddings` on GPU.

### PR DESCRIPTION
**Pull request by unverbraucht**
## Summary                                                                                                                                                                                                         
                                                                                                                                                                                                                   
Two related changes for using more embedding/reranker models on OpenArc:                                                                                                                                           
                                                                                                                                                                                                                   
- **`feat(embed)`**: dispatch pooling mode from sentence-transformers metadata so encoder-style embedders (bge-m3, e5, MiniLM, etc.) return correct vectors instead of last-token of the final non-pad position. Closes #89 .                                                                                                                                                                                              
- **`docs`**: add a working OpenVINO IR conversion recipe for Qwen3-Reranker that documents the Python-API path (the `optimum-cli export openvino` route silently truncates the XML to 0 bytes for this model).    
                                                                                                                                                                                                                   
Both are independent and low-risk; bundling per maintainer OK.                                                                                                                                                     
                                                                                                                                                                                                                   
## Embedding pool dispatch                                                                                                                                                                                         
                                                                                                                                                                                                                   
`Optimum_EMB.generate_embeddings` previously hardcoded `last_token_pool`. That's right for `Qwen3-Embedding-*` (decoder, last-token) but silently wrong for the much larger family of sentence-transformers encoders:                                                                                                                                                                                                                 
                                                                                                                                                                                                                   
| Model family | Correct pool | Old behavior |                                                                                                                                                                     
|---|---|---|                                                                                                                                                                                                      
| `Qwen3-Embedding-*` | last | ✅ |                                                                                                                                                                                
| `BAAI/bge-*` | **cls** | ❌ silently wrong |                                                                                                                                                                     
| `sentence-transformers/*`, `intfloat/multilingual-e5-*` | **mean** | ❌ silently wrong |                                                                                                                         
                                                                                                                                                                                                                   
The fix matches sentence-transformers' own precedence:                                                                                                                                                             
                                                                                                                                                                                                                   
1. **`runtime_config.pool_mode`** override (`"cls" | "mean" | "last"`). Unknown values raise at load time so typos don't fall back silently.                                                                       
2. **`<model_path>/1_Pooling/config.json`** auto-detect (`pooling_mode_cls_token` → cls, `pooling_mode_mean_tokens` → mean).                                                                                       
3. Default **`"last"`** — preserves Qwen3-Embedding behavior. No-op for existing users.                                                                                                                            
                                                                                                                                                                                                                   
### Verification                                                                                                                                                                                                   

- 14 unit tests covering each pool, the autodetect fallback chain, the runtime override, and unknown-mode rejection.
- Integration test loads the converted bge-m3 OV IR, asserts `pool_mode == "cls"` and a 1024-dim unit-normed vector.
- End-to-end against the PyTorch reference: `cos(ov, pt) > 0.999` for bge-m3.
- Live serving on GPU via `/v1/embeddings` returning correct vectors for bge-m3 and `/v1/rerank` for `qwen3-4b-reranker`.

## Qwen3-Reranker conversion docs

`docs/openvino_qwen3.md` walks through converting `Qwen/Qwen3-Reranker-{0.6B,4B}` to OpenVINO IR with INT8 weight compression via the `OVModelForCausalLM` Python API, which is the only path that produced a usable model end-to-end. The CLI path (`optimum-cli export openvino`) currently exits 0 while writing a 0-byte `openvino_model.xml` and a stub `.bin`; happy to file that separately upstream if useful.

## Test plan

- [x] `pytest src/tests/test_optimum_emb_unit.py` — all 14 pass
- [x] `pytest src/tests/test_optimum_emb_integration.py` — passes when bge-m3 / Qwen3-Embedding OV IRs are present, skips cleanly otherwise
- [x] `openarc serve` with `qwen3-4b-reranker` and `bge-m3` registered, exercised via `/v1/embeddings` and `/v1/rerank` against a live GPU
- [x] Cosine match vs PyTorch reference on bge-m3 (>0.999)
- [ ] No-op confirmation on existing `Qwen3-Embedding-*` deployments — requires a maintainer with that exact model loaded; default path is unchanged so behavior should be identical.

### CODE REVIEW

**Comment by SearchSavior** [comment]
@unverbraucht Thanks for submitting the PR! Great work.

There are some issues, mostly around conventions and following codebase patterns.

This PR infers the embedding task from `config.json` metadata; this approach may work with some models, but `OVModelForFeatureExtraction` covers many different versions of transformers making `config.json` somewhat unreliable. Better to be more specific as it helps define the codepath from request --> inference concretely making the whole addition highly extensible.

`runtime_config` was introduced to allow injecting OpenVINO runtime arguments as kwargs (anything you would pass into `openvino.Core`); this field has no default so the user can choose optimizations based on their hardware. As an entrypoint for defining the type of feature extraction task, this would be incorrect. Instead expect users to supply `model_type`; in OpenArc `ModelType` enum defines a codepath from the shared queuing/worker system into the expected unique inference code.

For this pr I would suggest updating `ModelType` `EMB = "emb" to

Following [L27](https://github.com/SearchSavior/OpenArc/pull/90/changes#diff-37df53918f78c907c096d442a6ff542eb3e3b7481855104a9b174e51c789e228R27)
```
OPTIMUM_EMB_MEAN = "emb_mean"
OPTIMUM_EMB_POOL = "emb_pool"
OPTIMUM_EMB_CLS = "emb_cls"
```

where `EMB = "emb"` becomes `OPTUMUM_EMB_MEAN = "emb_mean"` since that's the current default strategy. 

Based on how the async management works now, this enables wiring everything into the worker system for queuing and inference and would be required to get this merged. so each would need an `infer_emb_` method in `worker_registry.py`.

Embedding backend has been behind for a while, and you have made important changes! Thanks for your work. Maybe there is a better way, but your changes following the pattern elsewhere in OpenArc can shape a better abstraction in the future.

I also see some issues in the documentation .md, but love the addition. Covering specific quantization scenarios is a great idea and hardly covered in the official docs vs whats possible with the framework, but I think the agent may have struggled a bit as some of it's descriptions are awkward and might have to do with sentence transformers specific arguments (or something, I can investigate later)
